lib/database/users.ts

```diff
@@ -0,0 +1,122 @@
+import { client } from "../db";
+import { User, userSchema } from "../../models/users";
+import { string, array, TypeOf } from "yup";
+
+/**
+ * Creates a user and enters it into the database
+ *
+ *
+ */
+const createUser = async (
+  id: string,
+  first_name: string,
+  last_name: string,
+  email: string,
+  role: string,
+  address: string,
+  phone_number?: string
+): Promise<User> => {
+  // basic query syntax. We will be using pg to work with postgres. Read more
+  // it here: https://node-postgres.com/
+  const query = {
+    text:
+      "INSERT INTO users(id, first_name, last_name, email, role, address, phone_number) VALUES($1, $2, $3, $4, $5, $6, $7)" +
+      "RETURNING  id, first_name, last_name, email, role, phone_number, address",
```

Comment:
Instead of calling returning, how about we use the id and just return `findUser(id)`. Should make things easier. And then on the top in the API handler, if this function return null, then we know that something failed and we can return a 500